rust/core-lib/src/tabs.rs

```diff
@@ -390,9 +390,23 @@ impl CoreState {
             None => return,
         };
 
-        let ed = &self.editors[&buffer_id];
+        let fin_text = if let Some(mut save_ctx) = self.make_context(view_id) {
+            let editor = save_ctx.editor.borrow();
+            let rope = editor.get_buffer();
+            if save_ctx.pre_save_newline() {
+                let mut text = rope.to_string();
+                text.push('\n');
```

Comment:
hmm: we really need to be using the platform specific newline character, here. We can get that from the config manager.

What I might actually do here is add a helper method to `EventContext`, like `append_newline`, that returns a new rope that we can write to disk.

Reply:
> hmm: we really need to be using the platform specific newline character, here. We can get that from the config manager.

Oops, right, was a little in my UNIX world here :)